Initium-Odp/src/com/universeprojects/miniup/server/dao/OdpDao.java

```diff
@@ -0,0 +1,80 @@
+package com.universeprojects.miniup.server.dao;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.server.domain.OdpDomain;
+import com.universeprojects.miniup.server.exceptions.DaoException;
+
+public abstract class OdpDao<T extends OdpDomain> {
```

Comment:
I think this class could benefit by having many of the QueryHelper methods wrapped in it, returning proper wrapped versions of the result sets.